lib/Core/Executor.cpp

```diff
@@ -4574,6 +4574,12 @@ void Executor::runFunctionAsMain(Function *f,
   int envc;
   for (envc=0; envp[envc]; ++envc) ;
 
+  bool entryFunctionHasArguments = false;
+  if (!f->arg_empty() && f->getName() != "main") {
+    entryFunctionHasArguments = true;
+    klee_warning("Entry function %s has arguments", f->getName().data());
```

Comment:
I have kept this warning, because even if lazy-init is not used, this warns the user that the entry function he chose has some arguments, therefore execution will most likely fail if they are accessed.